testing/bin/subscription_tests.rs

```diff
@@ -0,0 +1,91 @@
+/*
+	Copyright 2019 Supercomputing Systems AG
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+//! Tests for the subscription interface functions.
+
+use codec::Decode;
+use frame_support::dispatch::DispatchInfo;
+use kitchensink_runtime::Runtime;
+use sp_keyring::AccountKeyring;
+use std::thread;
+use substrate_api_client::{
+	rpc::JsonrpseeClient, Api, AssetTipExtrinsicParams, GenericAddress, StaticEvent, XtStatus,
+};
+
+/// Check out frame_system::Event::ExtrinsicSuccess:
+#[derive(Decode)]
+struct ExtrinsicSuccess {
+	_dispatch_info: DispatchInfo,
+}
+
+impl StaticEvent for ExtrinsicSuccess {
+	const PALLET: &'static str = "System";
+	const EVENT: &'static str = "ExtrinsicSuccess";
+}
+
+#[tokio::main]
```

Comment:
I did not notice this before, we need the async runtime because jsonrpsee uses it internally?

Reply:
Yes, exactly.